MyOnboarding/src/MyOnboardingApp.Services/Services/UpdateItemService.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Threading.Tasks;
+using MyOnboardingApp.Contracts.Generators;
+using MyOnboardingApp.Contracts.Models;
+using MyOnboardingApp.Contracts.Repository;
+using MyOnboardingApp.Contracts.Services;
+using MyOnboardingApp.Contracts.Validation;
+
+namespace MyOnboardingApp.Services.Services
+{
+    internal class UpdateItemService: IUpdateItemService
+    {
+        private readonly IValidator<TodoListItem> _validator;
+        private readonly ITodoListRepository _repository;
+        private readonly IDateTimeGenerator _dateTimeGenerator;
+
+
+        public UpdateItemService(ITodoListRepository repository, IDateTimeGenerator dateTimeGenerator, IValidator<TodoListItem> validator)
+        {
+            _dateTimeGenerator = dateTimeGenerator;
+            _repository = repository;
+            _validator = validator;
+        }
+
+
+        public async Task<IResolvedItem<TodoListItem>> EditItemAsync(TodoListItem item)
+        {
+            var itemToReplace = CompleteItemToReplace(item);
```

Comment:
Well, to have a proper output from this method, it needs to get the original item from the database, doesn't it?  I mean, where else will you get proper `CreationTime` and any other potential properties you don't want to let through `PUT`? 
Why did your tests fail you here? (start with fixing tests first, of course 👼 )